src/test/java/edu/cornell/kfs/fp/batch/CreateAccountingDocumentReportItemTest.java

```diff
@@ -0,0 +1,93 @@
+package edu.cornell.kfs.fp.batch;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.kuali.kfs.sys.KFSConstants;
+
+class CreateAccountingDocumentReportItemTest {
+    
+    private static final String FILE_NAME = "foo.xml";
+    private static final String DEFAULT_WARNING_MESSAGE = "Any warning message will do";
+    private static final String DOCUMENT_TYPE_DV = "DV";
+    private static final String DOCUMENT_TYPE_DI = "DI";
+    private static final String DOCUMENT_TYPE_SB = "SB";
+    private static final String TWO_SPACES = KFSConstants.BLANK_SPACE + KFSConstants.BLANK_SPACE;
+    
+    CreateAccountingDocumentReportItem reportItem;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        reportItem = new CreateAccountingDocumentReportItem(FILE_NAME);
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, StringUtils.EMPTY));
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DI, TWO_SPACES));
+        reportItem.getDocumentsSuccessfullyRouted().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, TWO_SPACES));
+        reportItem.getDocumentsSuccessfullyRouted().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DI, StringUtils.EMPTY));
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        reportItem = null;
+    }
+    
+    private CreateAccountingDocumentReportItemDetail createCreateAccountingDocumentReportItemDetail(String documentType, String warningMessage) {
+        CreateAccountingDocumentReportItemDetail detail = new CreateAccountingDocumentReportItemDetail();
+        detail.setDocumentType(documentType);
+        detail.setWarningMessage(warningMessage);
+        return detail;
+    }
+    
+    private void addWarningDetails() {
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, DEFAULT_WARNING_MESSAGE));
+        reportItem.getDocumentsInError().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DI, DEFAULT_WARNING_MESSAGE));
+        reportItem.getDocumentsSuccessfullyRouted().add(createCreateAccountingDocumentReportItemDetail(DOCUMENT_TYPE_DV, DEFAULT_WARNING_MESSAGE));
+    }
+
+    @Test
+    public void testGetAllDocumentDetails() {
+        assertEquals(4, reportItem.getAllDocumentDetails().size());
+    }
+    
+    @Test
+    public void testGetDocumentDetailsWithWarningsNoWarnings() {
+        assertEquals(0, reportItem.getDocumentDetailsWithWarnings().size());
+    }
+    
+    @Test
+    public void testGetDocumentDetailsWithWarningsWithWarnings() {
```

Comment:
The "...WithWarningsWithWarnings" suffix is a little confusing, as well as the "NoWarnings"-suffixed equivalent. Could those two test methods be renamed for clarity?

Reply:
I adjusted the names, I hope it is better